sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowConverters.scala

```diff
@@ -158,7 +158,11 @@ private[sql] object ArrowConverters extends Logging {
             rowCountInLastBatch < maxRecordsPerBatch)) {
           val row = rowIter.next()
           arrowWriter.write(row)
-          estimatedBatchSize += row.asInstanceOf[UnsafeRow].getSizeInBytes
+          estimatedBatchSize += (row match {
```

Comment:
This is fine for now. For the record there are far better ways for determining the output size. Basically you can keep track of the schema size and the actual size of the vector root.